libs/encryption-sdk/src/main/java/org/opensearch/encryption/frame/TrimmingStream.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.encryption.frame;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Trims content from a given source range to a target range.
+ */
+public class TrimmingStream extends InputStream {
+
+    private final long sourceStart;
+    private final long sourceEnd;
+    private final long targetStart;
+    private final long targetEnd;
```

Comment:
While using streams we always have positions rather than lengths. It seems more convenient for callers to provide positions instead of making them calculate lengths.